lib/diagnostic.js

```diff
@@ -16,6 +30,16 @@ const min = function (previous, current) {
 // `let total = array.reduce(sum);`
 const sum = function (previous, current) {
 
+  // zero out current value
+  current = 0
+
+  // use an accumulator
+  for (let i = 0; i < previous.length; i++) {
```

Comment:
This function will be passed to a `reduce` function, so there is no need to actually use a loop in here!